src/core/execute.c

```diff
@@ -116,7 +130,8 @@ void exec_context_tty_reset(const ExecContext *context, const ExecParameters *p)
 
         if (p && p->stdin_fd >= 0)
                 fd = p->stdin_fd;
```

Comment:
if we have an stdin_fd passed in, then we should also have a similar logic, i.e. if isatty() fails on the fd we should exit cleanly without doing anything.